### PR TITLE
Add OptiFine

### PR DESCRIPTION
**Pull request by LostOfThought**
Needs work, however all initial filenames and SHA256's are avalible as of 05-02-17.

### CODE REVIEW

File: mods/optifine.json [hunk 1]
```diff
@@ -0,0 +1,4775 @@
+{
+    "format": 1,
+    "name": "OptiFine",
+    "authors": ["sp614x"],
+    "disc": "Minecraft optimization mod",
```
Comment (Forkk) [inline mods/optifine.json]: This should be `desc` instead of `disc`.

File: mods/optifine.json [hunk 2]
```diff
@@ -0,0 +1,4775 @@
...
+    "discLong": "This mod adds support for HD textures and a lot of options for better looks and performance. Doubling the FPS is common.",
```
Comment (Forkk) [inline mods/optifine.json]: There is no `descLong`. Long descriptions go in `desc`. There may be a `shortDesc` in the future, but not yet.

File: mods/optifine.json [hunk 3]
```diff
@@ -0,0 +1,4775 @@
...
+    "homepage": "http://optifine.net",
+    "license": {
+        "type": "custom",
+        "redistrubution": "false",
+        "text": "The mod OptiFine is Copyright © 2013 by sp614x and the intellectual property of the author. It may be not be reproduced under any circumstances except for personal, private use as long as it remains in its unaltered, unedited form. It may not be placed on any web site or otherwise distributed publicly without advance written permission. Use of this mod on any other website or as a part of any public display is strictly prohibited and a violation of copyright."
+    },
+    "versions": [
+        {
+            "name": "LagSpikeOfDeath_1.7.2.zip",
```
Comment (Forkk) [inline mods/optifine.json]: The name field for a version should include the mod's version number only, not the name of the mod.

File: mods/optifine.json [hunk 4]
```diff
@@ -0,0 +1,4775 @@
...
+            "mcvsn": ["beta_TODO"],
+            "hash": { "type": "sha256", "digest": "83fe6b5294161eb33b2f18f32a4af5077fd493a6047b1278ac448475d1fe159d"},
+            "urls": [
+                { "type": "archived", "url": "TODO", "desc": "TODO" },
```
Comment (Forkk) [inline mods/optifine.json]: If we haven't yet archived the file, this entry should be excluded.

File: mods/optifine.json [hunk 5]
```diff
@@ -0,0 +1,4775 @@
...
+                { "type": "original", "url": "TODO", "desc": "TODO" },
```
Comment (Forkk) [inline mods/optifine.json]: Same here. If there is no original direct download, this entry should be excluded from the list of URLs.

File: mods/optifine.json [hunk 6]
```diff
@@ -0,0 +1,4775 @@
+{
+    "format": 1,
+    "name": "OptiFine",
+    "authors": ["sp614x"],
+    "disc": "Minecraft optimization mod",
+    "discLong": "This mod adds support for HD textures and a lot of options for better looks and performance. Doubling the FPS is common.",
+    "homepage": "http://optifine.net",
+    "license": {
```
Comment (Forkk) [inline mods/optifine.json]: We don't have a license or homepage field in the schema. Maybe we should add one, though.

**Comment by Forkk** [comment]
Another thing, which I missed in the review (I don't think I can go back and add it) is that the `desc` field of URLs is optional, so you should omit it instead of setting it to "TODO" everywhere.

**Comment by LostOfThought** [comment]
I'll update when I get back home. I only made the pr as to at least have a starting point with some initial seed values even if not entirely proper.

File: mods/optifine.json (hunk 3 above)
Comment (LostOfThought) [inline mods/optifine.json]: So will we eventually have a filename field? Yes a filehash will always be an absolute for between versions, however even simple, not entirely sensible filename changes (such as upper to lowercase) can be relevant.

File: mods/LagSpikeOfDeath.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{
+    "format": 1,
+    "name": "LagSpikeOfDeath",
+    "authors": ["sp614x"],
+    "desc": "Minecraft optimization mod",
+    "versions": [
+        {
+            "name": "TODO_NAME",
```
Comment (Forkk) [inline mods/LagSpikeOfDeath.json]: This should be `1.7.2` for this version, and similar for the other versions.

File: mods/LagSpikeOfDeath.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+            "filename": "LagSpikeOfDeath_1.7.2.zip",
+            "mcvsn": ["Beta_TODO_VER"],
```
Comment (Forkk) [inline mods/LagSpikeOfDeath.json]: This should be filled in before these mods are merged.

File: mods/LagSpikeOfDeath.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
+{
+    "format": 1,
+    "name": "LagSpikeOfDeath",
+    "authors": ["sp614x"],
+    "desc": "Minecraft optimization mod",
+    "versions": [
+        {
+            "name": "TODO_NAME",
+            "filename": "LagSpikeOfDeath_1.7.2.zip",
```
Comment (Forkk) [inline mods/LagSpikeOfDeath.json]: I've just realized this field isn't in the spec, but I plan to add it, so keep it. :)

**Pull request closed without merging**